src/processutils.js

```diff
@@ -0,0 +1,80 @@
+const { spawn } = require('child_process')
+
+class ProcessUtils {
+  /**
+   * @typedef {Object} ProcessResult
+   * @property {number} code
+   * @property {number} signal
+   * @property {string} stdout
+   * @property {string} stderr
+   */
+
+  /**
+   * Run process returning a the handle and a with the result promise
+   * @param {*} command
+   * @param {*} args
+   * @param {*} options
+   * @returns {[ChildProcess, Promise<ProcessResult>]}
+   */
+  static runProcessAsync(command, args, options = {}) {
+    console.log(`${command} ${args.join(' ')}`)
+    let cmd = spawn(command, args, {
+      env: options.env,
+      detached: options.detached
+    })
+
+    let promise = new Promise((resolve, reject) => {
+      if (options.timeout) {
+        setTimeout(() => {
+          reject(new Error('Timeout'))
+          cmd.kill() // does not terminate the node process in the shell
+        }, options.timeout)
+      }
+
+      // Read stdout
+      let stdoutPromise = new Promise(resolve => {
+        let stdData = []
```

Comment:
Vuln to buffer overflow crashes if we do not have enough memory, buy maybe we dont care if we want this kind of API?